packages/@sanity/default-layout/src/components/DefaultLayout.js

```diff
@@ -120,13 +120,16 @@ export default withRouterHOC(
         }
       })
 
-
       return (
         <div
           className={`${styles.defaultLayout} ${mobileMenuIsOpen ? styles.mobileMenuIsOpen : ''}`}>
           {this.state.showLoadingScreen && (
             <div
-              className={this.state.loaded ? styles.loadingScreenLoaded : styles.loadingScreen}
+              className={
+                this.state.loaded && document.visibilityState == 'hidden'
```

Comment:
Shouldn't this be `this.state.loaded || document.visibilityState == 'hidden'` ? Also keep in mind that a change in document.visibilityState will _not_ trigger a re-render.

Reply:
I believe you are right - should be an `OR` here, not `AND`. I don't _think_ re-rendering on visibility change is needed, but maybe I'm not thinking this through fully.

Reply:
Agreed.